src/AzureOpenAIProxy.Management/Services/ModelService.cs

```diff
@@ -55,10 +92,60 @@ public async Task DeleteOwnerCatalogAsync(Guid catalogId)
         await db.SaveChangesAsync();
     }
 
+    public async Task<OwnerCatalog> GetOwnerCatalogAsync(Guid catalogId)
+    {
+        var result = await db.OwnerCatalogs.FindAsync(catalogId);
+
+        if (result is null)
+        {
+            return null!;
+        }
```

Comment:
Since this method has a non-null return, returning `null!` will break expectations of the caller (since that forces `null` to be allowed).

Either make the return type nullable, `Task<OwnerCatalog?>` or raise an exception that you catch in the caller